app/celery/nightly_tasks.py

```diff
@@ -334,3 +335,42 @@ def save_daily_notification_processing_time(bst_date=None):
             messages_within_10_secs=result.messages_within_10_secs,
         )
     )
+
+
+@notify_celery.task(name="delete-oldest-quarter-of-unneeded-notification-history")
+def delete_oldest_quarter_of_unneeded_notification_history():
+    # This retention limit is hardcoded for the moment. It is picked from
+    # https://github.com/alphagov/notifications-aws/blob/main/decisions/2022-12-01-notification-history-retention-period.md
+    # and can be increased in the future when we move 3 months into the next financial year
+    retention_limit = datetime(2022, 4, 1)
```

Comment:
Yes, I think it should be 2023 and I must have picked 2022 because that is what the ADR referred to (but as you point out, the ADR was written last financial year).

I will update it (and same with the relevant tests that check the around the time of the retention limit) and rebase the change in.

Reply:
I didn't end up rebasing the commit in, but left it as a separate commit for easy readability of my change.